tests: kernel/pipe_api: skip zero size test if KERNEL_COHERENCE

dcpleung opened the pull request:
Zero size buffer pipes are not currently supported if memory is not coherent between CPUs (CONFIG_KERNEL_COHERENCE=y) due to possibility of buffers being in incoherent memory. So skip the zero size test for now.

npitre reviewed (commented):
You can't just revert commit 0c0644bd6817..

Zero-sized pipes are simply not supported on systems where
CONFIG_KERNEL_COHERENCE=y.

The actual fix is to filter out the test when CONFIG_KERNEL_COHERENCE=y.

dcpleung committed:
tests: kernel/pipe_api: skip zero size test if KERNEL_COHERENCE

Zero size buffer pipes are not currently supported if memory is
not coherent between CPUs (CONFIG_KERNEL_COHERENCE=y) due to
possibility of buffers being in incoherent memory. So skip
the zero size test for now.

Signed-off-by: Daniel Leung <daniel.leung@intel.com>

andyross reviewed (approved):
Is the issue just the input/output arrays right here?  Or is there something internal to k_pipe that shares stack memory?  Might be worth an assert there if so, or even file an issue to get it fixed.
